### PR TITLE
drivers: flash: stm32 ospi adapt lines on writes

### PR DESCRIPTION
**Pull request by GeorgeCGV**
Correct used lines based on write opcode regardless of the data mode. The write opcode is set during init for all modes and can be overwritten in DT.

Simplify the switch, allow usage of different PP opcodes in OPI mode, and add `PP_1_1_2`.

### CODE REVIEW

**Comment by aescolar** [comment]
Bulk changing the milestone from everything which did not make it to the v3.7.0 freeze deadline
and which is not labeled as a bug or documentation (or does not appear clearly as only containing
a bug fix or a documentation change).
If you disagree about this change please add the milestone again.

Please note that until rc2 only PRs containing bug fixes, docs and tests can be merged.
After that and until release only bug fixes and docs changes.

Exceptions require approval by the release team and a vote by the TSC.
https://docs.zephyrproject.org/latest/project/release_process.html#stabilization-phase

**Comment by aescolar** [comment]
@erwango @GeorgeCGV If this is a bugfix, please add the "bug" label.
In that case you believe this should be considered for 3.7 you are welcome to change the milestone back

**Comment by GeorgeCGV** [comment]
@aescolar it is not a critical bug fix. It doesn't look like anyone uses non opi write opcode in opi mode or `SPI_NOR_CMD_PP_1_1_2` write opcode in other modes.

**Comment by MaureenHelm** [comment]
@erwango please take a look

**Review by FRASTM — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**
